lib/order.rb

```diff
@@ -0,0 +1,70 @@
+require_relative 'menu'
+require 'twilio-ruby'
+
+class Order
+
+  attr_reader :basket, :menu
+
+  def initialize(menu = Menu.new)
+    @menu = menu
+    @basket = []
+    @total = 0
+    @confirmed = false
+    time = (Time.now+(60*60))
+    @confirmation_text = "Thank you! Your order was placed and will be delivered before #{time.strftime("%H")}:#{time.strftime("%M")}. You ordered:"
+  end
+
+  def add_item(item, quantity = 1)
+    check_exists(item)
+    @confirmation_text += " #{quantity} portion(s) of #{item},"
+    retrieve_price(item)
+    recalculate_total(quantity)
+    basket << {item: item, quantity: quantity}
+  end
+
+  def total
+    @total
+  end
+
+  def confirmed?
+    @confirmed
+  end
+
+  def confirm
+    confirm_and_text
+  end
+
+  private
+
+  def check_exists(item)
+    item_found = false
+    @menu.list.each do |x|
+      if x.name == item
+        item_found = true
+      end
+    end
+    fail "Item on not on the menu" unless item_found
+  end
+
+  def retrieve_price(item)
+    item_to_find = @menu.list.select { |y| y.name == item }
+    @item_cost = item_to_find[0].price
```

Comment:
It's dangerous to introduce arbitrary state into your objects.  `@item_cost` does not make sense as an instance variable of `Order`.  Instead, return relevant methods from your methods and allow the caller to decide what to do with the result.